### PR TITLE
Add optional inflow visualization

### PR DESCRIPTION
**Pull request by rvanasa**
This PR adds a `show_inflow` option to the `build_sounding()` function which visualizes the CAPE, 3CAPE and CIN at each level of the atmosphere. 

The main use case is to quickly evaluate how strongly a storm is elevated or surface-based given different amounts of background forcing. The [effective inflow layer](https://journals.ametsoc.org/view/journals/mwre/148/8/mwrD200013.xml) is an approximation of the lowest contiguous layer of inflow shown in this visualization. 

Feel free to suggest or make any changes to this design.

Context: [Bluesky thread](https://bsky.app/profile/ryanvandersmith.com/post/3ldkhf2f5fs2u).

**Examples:**

2023 Yuma, CO EF3:

image

2024 messy MCS / tornado bust near Lufkin, TX. Note the secondary inflow layer from 700-500mb:

image

2024 Shreveport, LA tornado associated with Hurricane Beryl:

image

2024 surprise Lafayette, CO landspout:

image

2024 Bedford, TX 2" hail event:

image

2022 QLCS near Lakeside, NE:

image

[2023 Akron, CO supercell](https://www.weather.gov/bou/tornadoes_june_21_2023) which produced 27 tornadoes:

image

Dark mode:

image

### CODE REVIEW

File: src/sounderpy/calc.py [hunk 1]
```diff
@@ -394,6 +395,14 @@ def interpolate(var,hgts,step):
             thermo['dparcel_p'] = ma.masked
             thermo['dparcel_T'] = ma.masked
 
+        #--- PARCEL PROFILES ---#
+        # ---------------------------------------------------------------
+
+        if self.include_all_parcels:
+            parcels = [parcelx(prof, pres=p[i].m, tmpc=T[i].m, dwpc=Td[i].m) for i in range(len(p))]
```
Comment (rvanasa) [inline src/sounderpy/calc.py:402]: Performance overhead is negligible at about 1-2% of the total runtime of `build_sounding()`.

**Comment by kylejgillett** [comment]
Thanks Ryan!

This look really neat. SounderPy is always for new ways to visualize or conceptualize things! 

I do have one idea that I'd like to investigate with this before we move forward... I wonder how this might look plotted in one of the vertical graphs in the bottom right. Maybe in place of theta & thetae? Those graphs are all up to 3km, so we would be missing anything above 3km but I think having it down next to some 0-3km kinematics might also provide some clues as to what kinematics the storm is actually "feeling" -- i.e., "an elevated storm may not be feeling the 99% streamwiseness below 1km". We'd be able to easily visualize this.

Would you be willing to try that?

**Comment by rvanasa** [comment]
That's a great idea! I went ahead and opened another PR (#34) which gives this a try. Looking forward to hearing your feedback whenever you get a chance to take a look. Happy holidays!

**Comment by kylejgillett** [comment]
The changes proposed by #33 were reworked for #34, which has been merged with `main`.

**Pull request closed without merging**